test/e2e/upgrades/apps/deployments.go

```diff
@@ -98,89 +82,66 @@ func (t *DeploymentUpgradeTest) Setup(f *framework.Framework) {
+	if deployment.Annotations[deploymentutil.RevisionAnnotation] != "2" {
+		framework.ExpectNoError(fmt.Errorf("Deployment %q doesn't have the required revision set", deploymentName))
 	}
 
-	// Store new replica set - should be the same after the upgrade.
-	t.newRS = rs
-	deployment, err = c.Extensions().Deployments(ns).Get(deployment.Name, metav1.GetOptions{})
+	deployment, err = deploymentClient.Get(deployment.Name, metav1.GetOptions{})
 	framework.ExpectNoError(err)
 	t.oldD = deployment
 }
 
-// Test checks whether the replica sets for a deployment are the same after an upgrade.
+// Test checks whether the replicasets for a deployment are the same after an upgrade.
 func (t *DeploymentUpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
 	// Block until upgrade is done
-	By(fmt.Sprintf("Waiting for upgrade to finish before checking replica sets for deployment %q", t.oldD.Name))
+	By(fmt.Sprintf("Waiting for upgrade to finish before checking replicasets for deployment %q", t.oldD.Name))
 	<-done
 
 	c := f.ClientSet
+	name := t.oldD.Name
+	deploymentClient := c.ExtensionsV1beta1().Deployments(t.oldD.Namespace)
 
-	deployment, err := c.Extensions().Deployments(t.oldD.Namespace).Get(t.oldD.Name, metav1.GetOptions{})
+	deployment, err := deploymentClient.Get(name, metav1.GetOptions{})
 	framework.ExpectNoError(err)
 	t.updatedD = deployment
 
-	By(fmt.Sprintf("Waiting for deployment %q to complete adoption", deployment.Name))
-	framework.ExpectNoError(framework.WaitForDeploymentStatus(c, deployment))
-
-	By(fmt.Sprintf("Checking that replica sets for deployment %q are the same as prior to the upgrade", t.updatedD.Name))
-	_, allOldRSs, newRS, err := deploymentutil.GetAllReplicaSets(t.updatedD, c.ExtensionsV1beta1())
-	framework.ExpectNoError(err)
-	if newRS == nil {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected a new replica set for deployment %q", t.updatedD.Name))
+	By(fmt.Sprintf("Checking UID to verify deployment %q survives upgrade", name))
+	if t.updatedD.UID != t.oldD.UID {
+		framework.ExpectNoError(fmt.Errorf("expected same deployment UID: %v got: %v", t.oldD.UID, t.updatedD.UID))
 	}
-	if newRS.UID != t.newRS.UID {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected new replica set:\n%#v\ngot new replica set:\n%#v\n", t.newRS, newRS))
-	}
-	if len(allOldRSs) != 1 {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		errString := fmt.Sprintf("expected one old replica set, got %d\n", len(allOldRSs))
-		for i := range allOldRSs {
-			rs := allOldRSs[i]
-			errString += fmt.Sprintf("%#v\n", rs)
-		}
-		framework.ExpectNoError(fmt.Errorf(errString))
-	}
-	if allOldRSs[0].UID != t.oldRS.UID {
-		By(t.spewDeploymentAndReplicaSets(newRS, allOldRSs))
-		framework.ExpectNoError(fmt.Errorf("expected old replica set:\n%#v\ngot old replica set:\n%#v\n", t.oldRS, allOldRSs[0]))
+
```

Comment:
You'd also want to check that the deployment didn't create more replicasets after upgrade.